SneakPeek/src/com/cyscorpions/dalejulian/sneakpeek/CategoryJSONSerializer.java

```diff
@@ -0,0 +1,85 @@
+package com.cyscorpions.dalejulian.sneakpeek;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONTokener;
+
+import android.content.Context;
+
+public class CategoryJSONSerializer {
+
+	private Context mContext;
+	private String mFileName;
+
+	public CategoryJSONSerializer(Context c, String f) {
+		this.mContext = c;
+		this.mFileName = f;
+	}
+
+	public ArrayList<SneakerCategory> loadCategories() throws JSONException,
+			IOException {
+		ArrayList<SneakerCategory> categories = new ArrayList<SneakerCategory>();
+		categories.add(new SneakerCategory("Basketball",
+				"Air Jordans, Lebrons, Kobes.."));
+		categories.add(new SneakerCategory("Running",
+				"Nike Air Max, Adidas Pure Boost, Under Armour.."));
+		categories.add(new SneakerCategory("Skateboarding",
+				"NikeSB, Emerica, DC Shoes.."));
+		categories.add(new SneakerCategory("Designer",
+				"Balenciaga, Maison Martin,  Saint Laurent.."));
+		BufferedReader reader = null;
+		try {
+			InputStream in = mContext.openFileInput(mFileName);
+			reader = new BufferedReader(new InputStreamReader(in));
+			StringBuilder jsonString = new StringBuilder();
+			String line = null;
+			while ((line = reader.readLine()) != null) {
+				jsonString.append(line);
+			}
+			JSONArray jsonArray = (JSONArray) new JSONTokener(
+					jsonString.toString()).nextValue();
+			for (int i = 0; i < jsonArray.length(); i++) {
```

Comment:
``` jsonArray.length() ```
Place the value of this to a variable, so that this for-loop does not iterate on calling this (length) function each time. :dart:  (Please apply to all others)